webdriver/tests/bidi/storage/set_cookie/set_cookie.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from webdriver.bidi.modules.network import NetworkStringValue
+from webdriver.bidi.modules.script import ContextTarget
+from webdriver.bidi.modules.storage import PartialCookie, BrowsingContextPartitionDescriptor
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.mark.parametrize(
+    "protocol",
+    [
+        "http",
+        "https",
+    ],
+)
+async def test_set_cookie_protocol(bidi_session, top_context, inline, origin, domain_value, protocol):
+    # Navigate to a page with a required protocol.
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=(inline("<div>foo</div>", protocol=protocol)), wait="complete"
+    )
+
+    source_origin = origin(protocol)
+    partition = BrowsingContextPartitionDescriptor(top_context["context"])
+
+    set_cookie_result = await bidi_session.storage.set_cookie(
+        cookie=PartialCookie(
+            name='foo',
+            value=NetworkStringValue('bar'),
+            domain=domain_value(),
+            secure=True
+        ),
+        partition=partition)
+
+    assert set_cookie_result == {
+        'partitionKey': {
+            'sourceOrigin': source_origin
+        },
+    }
+
+    # Assert the cookie is actually set.
+    actual_cookies = await bidi_session.storage.get_cookies(partition=partition)
+    assert actual_cookies == {
+        'cookies': [
+            {
+                'domain': 'web-platform.test',
```

Comment:
Looks like the linting job is not happy that you use the web-platform domain here directly, maybe we could use some fixture instead.

Reply:
Yes, but usually I write tests in gecko repo and run lint there